src/main/java/net/sf/jabref/logic/cleanup/ISSNCleanup.java

```diff
@@ -0,0 +1,33 @@
+package net.sf.jabref.logic.cleanup;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.model.FieldChange;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class ISSNCleanup implements CleanupJob {
+
+    private static final Pattern ISSN_PATTERN_NODASH = Pattern.compile("^(\\d{4})(\\d{3}[\\dxX])$");
+    @Override
+    public List<FieldChange> cleanup(BibEntry entry) {
+        Optional<String> issn = entry.getFieldOptional("issn");
+        if (!issn.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        Matcher matcher = ISSN_PATTERN_NODASH.matcher(issn.get());
```

Comment:
Sure, but it can be cleaned up as opposed to all other cases.

Maybe I was focusing too much on your example, but to not set the field
based on a missing dash seemed quite stupid to me. Better then to add it.
Hence the questions how to handle that.

Still, I do not really see it happening at the moment though.